src/Microsoft.Azure.ServiceBus.MessageId/MessageIdPlugin.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.ServiceBus.Core;
+
+namespace Microsoft.Azure.ServiceBus.MessageId
+{
+    /// <summary>Generate Azure Service Bus <see cref="Message"/> id.</summary>
+    public class MessageIdPlugin : ServiceBusPlugin
+    {
+        private readonly Func<string> messageIdGenerator;
```

Comment:
@nemakam why would I do that? You're registering this plugin to do the work, not to invoke the generator directly. Not to mention that the invoking code already has access to the generator func.